src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -215,34 +210,56 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
         /// <summary>
         /// This function is used to get the final request uri after the lro has completed.
         /// </summary>
-        private string? GetFinalUri()
+        private string? GetFinalUri(Response response)
         {
             // Set final uri as null if the response header doesn't contain "Operation-Location" or "Azure-AsyncOperation".
             if (_headerSource is not (HeaderSource.OperationLocation or HeaderSource.AzureAsyncOperation))
             {
                 return null;
             }
 
-            // Set final uri as null if original request is a delete method.
-            if (_requestMethod == RequestMethod.Delete)
+            if (_finalStateVia == OperationFinalStateVia.AzureAsyncOperation && _requestMethod == RequestMethod.Post)
```

Comment:
Why was this check added? I don't see any old code like this, and looking at the other bug this should close, it seems to have nothing to do with the custom Azure async location header. Perhaps it needs an explanation comment like the old code did here.